### PR TITLE
Provides support

### PR DESCRIPTION
**Pull request by erebus1**

### CODE REVIEW

**Comment by erebus1** [comment]
We need somehow register types, which has fields marked as `@provides` to patch sdl ...

File: graphene_federation/service.py [hunk 1]
```diff
@@ -46,6 +53,7 @@ def get_sdl(schema, custom_entities):
     string_schema = pattern.sub(" ", string_schema)
 
     for entity_name, entity in custom_entities.items():
+        string_schema = _mark_provides(entity_name, entity, string_schema, schema.auto_camelcase)
```
Comment (erebus1) [inline graphene_federation/service.py]: We need to go through all types, not only those decorated with `@key`

File: integration_tests/service_c/src/schema.py [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
+    age = external(Int())
 
     def resolve_uppercase_email(self, info):
         return self.primary_email.upper() if self.primary_email else self.primary_email
 
+    def resolve_age(self, info):
+        return 18
+
 
 @key(fields='id')
```
Comment (erebus1) [inline integration_tests/service_c/src/schema.py]: `@key` decorator was added in another PR
Thus test passed
we need to use type without `@key` decorator

File: integration_tests/service_c/src/schema.py [hunk 3]
```diff
@@ -1,22 +1,26 @@
...
 class Article(ObjectType):
     id = Int(required=True)
     text = String(required=True)
-    author = Field(lambda: User)
+    author = provides(Field(lambda: User), fields='age', parent_type=lambda: Article)
```
Comment (erebus1) [inline integration_tests/service_c/src/schema.py]: Or add decorator for class, to mark as a class that can provide? :thinking:

**Comment by petebrowne** [comment]
What's the status of this? Do you need any input or help?

File: integration_tests/service_c/src/schema.py (hunk 2 above)
Comment (erebus1) [reply integration_tests/service_c/src/schema.py:19]: resolved

File: graphene_federation/service.py (hunk 1 above)
Comment (erebus1) [reply graphene_federation/service.py]: resolved

**Comment by erebus1** [comment]
@petebrowne
Feel free to check the new version on PyPI

**Comment by petebrowne** [comment]
Will do, thanks!

**Pull request merged**